client/concordclient/src/concord_client.cpp

```diff
@@ -35,13 +35,13 @@ void ConcordClient::send(const bft::client::WriteConfig& config,
 void ConcordClient::subscribe(const SubscribeRequest& request,
                               const std::unique_ptr<opentracing::Span>& parent_span,
                               const std::function<void(SubscribeResult&&)>& callback) {
-  if (not stop_subscriber_) {
+  if (subscriber_) {
     LOG_ERROR(logger_, "subscription already in progress - unsubscribe first");
     throw SubscriptionExists();
   }
 
   stop_subscriber_ = false;
-  subscriber_ = std::thread([&] {
+  subscriber_ = std::make_unique<std::thread>([&] {
     while (not stop_subscriber_) {
```

Comment:
Why are we still checking the flag `stop_subscriber_` here, instead of checking the pointer like we do above?

Reply:
The pointer represents the thread. The while-loop is within the thread so it would always be true. In this while-loop we specifically want to check if the "outside" (the gRPC server) wants us to stop.

Reply:
Got it, thanks for the explanation.